x-pack/test/observability_ai_assistant_api_integration/tests/knowledge_base/knowledge_base_status.spec.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../common/ftr_provider_context';
+import { SUPPORTED_TRAINED_MODELS } from '../../../functional/services/ml/api';
+
+export default function ApiTest({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+  const ml = getService('ml');
+  const TINY_ELSER = {
+    ...SUPPORTED_TRAINED_MODELS.TINY_ELSER,
+    id: SUPPORTED_TRAINED_MODELS.TINY_ELSER.name,
+  };
+  const KNOWLEDGE_BASE_API_URL = `/internal/observability_ai_assistant/kb`;
+
+  describe(`${KNOWLEDGE_BASE_API_URL}/status`, () => {
+    before(async () => {
+      await ml.api.importTrainedModel(TINY_ELSER.name, TINY_ELSER.id);
+      await ml.api.assureMlStatsIndexExists();
+
+      await supertest
+        .post(`${KNOWLEDGE_BASE_API_URL}/setup`)
+        .set('kbn-xsrf', 'foo')
+        .expect(200)
+        .then((response) => {
+          expect(response.body).to.eql({});
+        });
+    });
+
+    after(async () => {
+      await ml.api.stopTrainedModelDeploymentES(TINY_ELSER.id, true);
+      await ml.api.deleteTrainedModelES(TINY_ELSER.id);
+      await ml.api.cleanMlIndices();
+      await ml.testResources.cleanMLSavedObjects();
```

Comment:
maybe a good candidate for extracting it out into a separate function?

Reply:
i separated these into helper functions as I think I'll need them for other tests